### PR TITLE
Fix id mismatch for units.

### PR DESCRIPTION
**Pull request by huwshimi**
Fix lookup of units if they were left uncommitted when a service was committed.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- add a service
- navigate to the uncommitted unit details
- deploy the service (leaving the unit uncommitted)
- navigate to the uncommitted unit details
- check that there are no errors in your console

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3257/
Test PASSed.

File: jujugui/static/gui/src/app/models/models.js [hunk 1]
```diff
@@ -1012,6 +1012,25 @@ YUI.add('juju-models', function(Y) {
...
+      @param {String} displayName The display name for a unit.
+      @return {Object} The matching unit with the matching display name.
+    */
+    getByDisplayName: function(displayName) {
+      var units = this.filter(function(unit) {
+        if (unit.displayName === displayName) {
+          return unit;
+        }
+      });
+      if (units.length > 0) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/models/models.js]: curious why there would be more than one unit which matches the display name? Could you add a comment to this effect.

File: jujugui/static/gui/src/test/test_model.js [hunk 2]
```diff
@@ -1084,6 +1084,14 @@ describe('test_model.js', function() {
       });
     });
 
+    it('can get a unit but displayName', function() {
```
Comment (hatched) [inline jujugui/static/gui/src/test/test_model.js]: by

**Comment by hatched** [comment]
I didn't know this was actually a problem, I had to go and test it without the patch first :) Good find!

I'm wondering if this is actually just covering up the problem though. I think that the real fix is to update these unit id's on deploy so that this problem doesn't exist at all.

File: jujugui/static/gui/src/app/models/models.js (hunk 1 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/models/models.js]: I should probably just update that to be `=== 0`

**Comment by huwshimi** [comment]
A better fix for this issue is over at: https://github.com/juju/juju-gui/pull/933

**Pull request closed without merging**